src/__main__.py

```diff
@@ -1,23 +1,30 @@
-
 import sys
 import webbrowser
+import os
 
-from flask import Flask, render_template, request
+from flask import Flask, render_template, request, redirect, url_for, send_from_directory
 from gevent.pywsgi import WSGIServer
 from PyQt5.QtWidgets import QApplication, QGridLayout, QLabel, QPushButton, QWidget
+from werkzeug.utils import secure_filename
+from werkzeug.middleware.shared_data import SharedDataMiddleware
 
+UPLOAD_FOLDER = 'uploads'
 APP_NAME = 'loft'
 HOST = '0.0.0.0'
 PORT = 2402
 
 gui = QApplication(sys.argv)
 app = Flask(APP_NAME)
 server = WSGIServer((HOST, PORT), app)
+app.config['UPLOAD_FOLDER'] = UPLOAD_FOLDER
+ALLOWED_EXTENSIONS = {'txt', 'pdf', 'png', 'jpg', 'jpeg', 'gif'}
```

Comment:
Ethan mentioned removing allowed extensions. Might prevent some issues like apple devices uploading their HEIC image format (not sure if thats what it is, I just know its weird)